### DISCUSSION TITLE
How can an admin impersonate as a customer

### DISCUSSION THREAD

**Discussion by ivanakcheurov** [saleor/saleor#12947] [Q&A]
Hi,
Thank you for building Saleor! Great product!

### Usecase
I am running a webshop on Saleor and have some custom integration to show additional dashboards to customers.
One of customers reported that she cannot see the dashboard in her account anymore while everything else was working good.
I needed a quick way to see what she can see. In WordPress and WooCommerce, I (as admin) used to impersonate as any other user and see what they would see in the shop (so including browsing MyAccount as if I were the other user).

### Question
Is there a way for admins to impersonate as other users in Saleor? So that I can see My Account of any user on the website like if I were that user but without knowing their password.

**Comment by timuric** [comment]
@ivanakcheurov that depends on your front end implementation, given that staff user has enough rights they are able to browse customers orders, but there are limitations, if you are using `me` query you would have to be actually logged in under customers account. You could implement impersonation with custom authorization provider using OIDC plugin in Saleor.
